File: src/main/java/antimattaermod/core/Energy/Machine/BlockAlloySmelter.java
package antimattaermod.core.Energy.Machine;

import net.minecraft.block.BlockContainer;
import net.minecraft.block.material.Material;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.World;

/**
 * @author C6H2Cl2
 */
public class BlockAlloySmelter extends BlockContainer {
    private final int tier;
    public BlockAlloySmelter(int tier) {
        super(Material.rock);
        this.tier = tier;
    }

    @Override
    public boolean onBlockActivated(World world, int x, int y, int z, EntityPlayer player, int side, float p_149727_7_, float p_149727_8_, float p_149727_9_) {
        return false;
    }

    @Override
    public TileEntity createNewTileEntity(World p_149915_1_, int p_149915_2_) {
        return new TileAlloySmelter(tier);
    }
}

File: src/main/java/antimattermod/core/Energy/Block/Machine/BlockAlloySmelter.kt
package antimattermod.core.Energy.Block.Machine

import antimattermod.core.AMMGuiHandler
import antimattermod.core.AntiMatterModCore
import antimattermod.core.AntiMatterModRegistry
import antimattermod.core.Energy.TileEntity.Machine.TileAlloySmelter
import c6h2cl2.YukariLib.Block.BlockWithTileEntity
import net.minecraft.block.material.Material
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.world.World

/**
 * @author C6H2Cl2
 */
class BlockAlloySmelter(private val tier: Int) : BlockWithTileEntity<TileAlloySmelter>(
        Material.rock, TileAlloySmelter::class, { world, meta -> TileAlloySmelter(tier)}, "amm.AlloySmelter.$tier"
) {
    init {
        setCreativeTab(AntiMatterModRegistry.tabMachines)
        setBlockName("alloySmelter_tier" + tier)
        setBlockTextureName(AntiMatterModCore.MOD_ID.toLowerCase() + ":alloySmelter_tier" + tier)
        setHardness(40f)
        setResistance(50f)
        setHarvestLevel("pickaxe", tier + 2)
    }

    override fun onBlockActivated(world: World?, x: Int, y: Int, z: Int, player: EntityPlayer?, side: Int, hitX: Float, hitY: Float, hitZ: Float): Boolean {
        player!!.openGui(AntiMatterModCore.INSTANCE, AMMGuiHandler.GuiID_AlloySmelter, world, x, y, z)
        return true
    }
}
